modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -3023,13 +3098,22 @@ private void scheduleTransactionRowAsyncCleanup(UUID txId, RowId rowId, Transact
 
         // Both normal cleanup and single row cleanup are using txsPendingRowIds map to store write intents.
         // So we don't need a separate method to handle single row case.
-        txManager.executeCleanupAsync(() ->
-                inBusyLock(busyLock, () -> storageUpdateHandler.handleTransactionCleanup(txId, txState == COMMITED, commitTimestamp))
-        ).exceptionally(e -> {
-            LOG.warn("Failed to complete transaction cleanup command [txId=" + txId + ']', e);
+        CompletableFuture<?> future = rowCleanupMap.compute(rowId, (k, v) -> {
```

Comment:
This part and generally exception handling flow is confusing.
First of all, how are we going to handle exceptions in 
```
 return validateOperationAgainstSchema(txId)
                                    .thenCompose(catalogVersion -> checkCleanup(rowIdLock.get1(), catalogVersion))
```
if checkCleanup will return the future completed exceptionally.